lib/AST/CompilerPlugin.cpp

```diff
@@ -188,9 +188,9 @@ void ASTContext::loadCompilerPlugins() {
     swift_ASTGen_getMacroTypes(getter, &metatypesAddress, &metatypeCount);
     ArrayRef<const void *> metatypes(metatypesAddress, metatypeCount);
     for (const void *metatype : metatypes) {
-      CompilerPlugin plugin(metatype, lib, *this);
-      auto name = plugin.getName();
-      LoadedPlugins.try_emplace(name, std::move(plugin));
+      auto plugin = new CompilerPlugin(metatype, lib, *this);
```

Comment:
Should this be allocated on the `ASTContext` since we never unload plugins?  Though, we could in theory unload them after type checking.